### PR TITLE
feat: [PLG-1377]: Added endpoint for subscription recommendations

### PR DESCRIPTION
**Pull request by thomas-grasman**
## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>
  
- Feature build: `trigger feature-build`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>
  
  You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`
  
- Compile: `trigger compile`
- CodeFormat: `trigger codeformat`
- MessageMetadata: `trigger messagecheck`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- runDockerizationCheck: `trigger dockerizationcheck`
- runAuthorCheck: `trigger authorcheck`
- Checkstyle: `trigger checkstyle`
- PMD: `trigger pmd`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/36309)

### CODE REVIEW

**Comment by diptiman-adak** [comment]
Quality gate Quality gate: **_Failed_** 
 


            
 #### 2 Issues 
* Code issue 1 Code issues 
* Anti-pattern 1 Anti-patterns 

 [Details](https://app.embold.io/organization/gh/harness/repositories/7752dbb3099d2dba3b5d37f5931a5e4c/pull-request/36309)

**Review by pranil-harness — APPROVED** [review]
> Reviewed 5 of 5 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/36309)* status: Some of the checks are too old. 5 file(s) to review. 1 of 2 approvals obtained.

**Review by pranil-harness — DISMISSED** [review]
> Reviewed 1 of 1 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/36309)* status: 5 file(s) to review. 1 of 2 approvals obtained.

File: 925-subscription-service/src/main/java/io/harness/subscription/services/impl/SubscriptionServiceImpl.java [hunk 1]
```diff
@@ -50,23 +62,89 @@
...
+        throw new InvalidRequestException(String.format("Cannot get recommendation for module %s,", moduleType));
+    }
+
+    EnumMap<UsageKey, Long> recommendedValues = new EnumMap<>(UsageKey.class);
+
+    LicenseType licenseType = latestLicense.getLicenseType();
+    Edition edition = latestLicense.getEdition();
+    if (licenseType.equals(LicenseType.TRIAL)) {
+      baseValues.forEach((key, value) -> {
+        double recommendedValue = Math.max(value, usage.get(key)) * 1.2d;
```
Comment (pranil-harness) [inline 925-subscription-service/src/main/java/io/harness/subscription/services/impl/SubscriptionServiceImpl.java:132]: 1.2d - magic number? make it a constatn

Comment (pranil-harness) [reply 925-subscription-service/src/main/java/io/harness/subscription/services/impl/SubscriptionServiceImpl.java:132]: Does 1.2 factor apply to current usage only or to max between value and usage?

Comment (thomas-grasman) [reply 925-subscription-service/src/main/java/io/harness/subscription/services/impl/SubscriptionServiceImpl.java]: On trial, It only takes 120% of current usage, as outlined in the PLG-1377 card.

**Review by pranil-harness — APPROVED** [review]
> Reviewed 1 of 1 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/36309)* status: 5 file(s) to review. 1. unresolved discussions. 1 of 2 approvals obtained.

**Review by hekimoglu010 — APPROVED** [review]
> Reviewed 4 of 5 files at r1, 1 of 1 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/36309)* status: 1. unresolved discussions.

**Review by devinder-harness — APPROVED** [review]

**Review by devinder-harness — APPROVED** [review]
> Reviewed 4 of 5 files at r1, 1 of 1 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/36309)* status: 1. unresolved discussions.

**Comment by thomas-grasman** [comment]
trigger messagecheck

**Pull request merged**